applications/StructuralMechanicsApplication/custom_elements/solid_elements/small_displacement_mixed_strain_displacement_element.cpp

```diff
@@ -0,0 +1,1278 @@
+#include "custom_utilities/constitutive_law_utilities.h"
+
+namespace Kratos
+{
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+Element::Pointer SmallDisplacementMixedStrainDisplacementElement::Create(
+    IndexType NewId,
+    NodesArrayType const& ThisNodes,
+    PropertiesType::Pointer pProperties) const
+{
+    return Kratos::make_intrusive<SmallDisplacementMixedStrainDisplacementElement>(NewId, GetGeometry().Create(ThisNodes), pProperties);
+}
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+Element::Pointer SmallDisplacementMixedStrainDisplacementElement::Create(
+    IndexType NewId,
+    GeometryType::Pointer pGeom,
+    PropertiesType::Pointer pProperties) const
+{
+    return Kratos::make_intrusive<SmallDisplacementMixedStrainDisplacementElement>(NewId, pGeom, pProperties);
+}
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+Element::Pointer SmallDisplacementMixedStrainDisplacementElement::Clone(
+    IndexType NewId,
+    NodesArrayType const& rThisNodes) const
+{
+    KRATOS_TRY
+
+    SmallDisplacementMixedStrainDisplacementElement::Pointer p_new_elem = Kratos::make_intrusive<SmallDisplacementMixedStrainDisplacementElement>(NewId, GetGeometry().Create(rThisNodes), pGetProperties());
+    p_new_elem->SetData(this->GetData());
+    p_new_elem->Set(Flags(*this));
+
+    // Currently selected integration methods
+    p_new_elem->SetIntegrationMethod(mThisIntegrationMethod);
+
+    // The vector containing the constitutive laws
+    p_new_elem->SetConstitutiveLawVector(mConstitutiveLawVector);
+
+    return p_new_elem;
+
+    KRATOS_CATCH("");
+}
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+void SmallDisplacementMixedStrainDisplacementElement::EquationIdVector(
+    EquationIdVectorType& rResult,
+    const ProcessInfo& rCurrentProcessInfo) const
+{
+    KRATOS_TRY
+
+    const auto& r_geometry     = GetGeometry();
+    const SizeType n_nodes     = r_geometry.PointsNumber();
+    const SizeType dim         = r_geometry.WorkingSpaceDimension();
+    const SizeType strain_size = mConstitutiveLawVector[0]->GetStrainSize();
+    const SizeType u_size      = dim * n_nodes;
+    const SizeType dof_size    = n_nodes * (dim + strain_size);
+
+    if (rResult.size() != dof_size){
+        rResult.resize(dof_size);
+    }
+
+    if (dim == 2) {
+        for (IndexType i_node = 0; i_node < n_nodes; ++i_node) {
+            const auto &r_node = GetGeometry()[i_node];
```

Comment:
You have the geometry already defined
```suggestion
            const auto &r_node = r_geometry[i_node];
```